packages/phone-number-privacy/signer/src/domain/domain.service.ts

```diff
@@ -1,49 +1,75 @@
-import { DisableDomainRequestBody } from '@celo/phone-number-privacy-common'
+import {
+  DomainRequestBody,
+  DomainStatusResponse,
+  ErrorMessage,
+} from '@celo/phone-number-privacy-common'
 import { Request, Response } from 'express'
-import * as core from 'express-serve-static-core'
+import { respondWithError } from '../common/error-utils'
+import { DOMAINS_COLUMNS } from '../database/models/domain'
 import { getDomain, setDomainDisabled } from '../database/wrappers/domain'
+import { Endpoints } from '../server'
 import { IDomainService } from './domain.interface'
-import { DOMAINS_COLUMNS } from '../database/models/domain'
-import { DomainStatusResponse } from './response/domainStatus.response'
 
 export class DomainService implements IDomainService {
-  constructor() {}
-
   public async handleDisableDomain(
-    request: Request<core.ParamsDictionary, {}, DisableDomainRequestBody>,
+    request: Request<{}, {}, DomainRequestBody>,
     response: Response
   ): Promise<void> {
     const logger = response.locals.logger
-    logger.info('Disabling domain', { domain: request.params.domain })
+    logger.info('Disabling domain', { domain: request.body.domain })
 
-    const disabled = request.body.value ?? true
-    const result = await setDomainDisabled(request.params.domain, disabled, logger)
-
-    if (result) {
-      response.sendStatus(200)
-    } else {
-      response.status(500).send({ message: 'Cannot disable the domain' })
-    }
+    return setDomainDisabled(request.body.domain, logger)
```

Comment:
We should first check that the domain exists in the db and is not already disabled. We can return different messages for both of these cases